compatible/src/main/java/io/seata/saga/proctrl/ProcessContext.java

```diff
@@ -97,4 +98,10 @@ public interface ProcessContext extends org.apache.seata.saga.proctrl.ProcessCon
      * @return the get instruction
      */
     <T extends Instruction> T getInstruction(Class<T> clazz);
+
+    /**
+     * Gets get lock.
+     * @return the lock of the current process context
+     */
+    ResourceLock getLock();
```

Comment:
The getLock method doesn't really fit a universal business process or template setup, so it's not quite suitable to include it in the interface.